internal/report/report.go

```diff
@@ -0,0 +1,183 @@
+package report
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/wg-policy-prototypes/policy-report/pkg/api/wgpolicyk8s.io/v1alpha2"
+
+	"github.com/kubewarden/audit-scanner/internal/constants"
+	policiesv1 "github.com/kubewarden/kubewarden-controller/pkg/apis/policies/v1"
+	admv1 "k8s.io/api/admission/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type PolicyReport struct {
+	v1alpha2.PolicyReport
+}
+type ClusterPolicyReport struct {
+	v1alpha2.ClusterPolicyReport
+}
+
+type RInterface interface {
+	metav1.Object
+	// GetType returns the type of the PolicyReport
+	GetType() ResourceType
+	// AddResult adds a PolicyReportResult to the policy report. This result
+	// includes policy info and resource under audit
+	AddResult(policiesv1.Policy, unstructured.Unstructured, *admv1.AdmissionReview, error)
+}
+
+// ResourceType Enum defined for PolicyReport
+type ResourceType = string
+
+// ReportType Enum
+const (
+	PolicyReportType        ResourceType = "PolicyReport"
+	ClusterPolicyReportType ResourceType = "ClusterPolicyReport"
+)
+
+// Status specifies state of a policy result
+const (
+	StatusPass  = "pass"
+	StatusFail  = "fail"
+	StatusWarn  = "warn"
+	StatusError = "error"
+	StatusSkip  = "skip"
+)
+
+func NewCPR(name string) *ClusterPolicyReport {
+	return &ClusterPolicyReport{
+		ClusterPolicyReport: v1alpha2.ClusterPolicyReport{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              PrefixNameClusterPolicyReport + name,
+				CreationTimestamp: metav1.Now(),
+			},
+			Summary: v1alpha2.PolicyReportSummary{
+				Pass:  0, // count of policies with requirements met
+				Fail:  0, // count of policies with requirements not met
+				Warn:  0, // not used for now
+				Error: 0, // count of policies that couldn't be evaluated
+				Skip:  0, // count of policies that were not selected for evaluation
+			},
+			Results: []*v1alpha2.PolicyReportResult{},
+		},
+	}
+}
+
+func NewPR(namespace *v1.Namespace) *PolicyReport {
```

Comment:
nitpicking: I would prefer a more explicit name: `NewPolicyReport`

Reply:
Done in https://github.com/kubewarden/audit-scanner/pull/40/commits/898d19140dc59093af5165a216f4595643d2e004